netmaster/daemon.go

```diff
@@ -405,3 +415,31 @@ func (d *daemon) runMasterFsm() {
 		}
 	}
 }
+
+func (d *daemon) restoreCache() {
+
+	//Restore ServiceLBDb and ProviderDb
+	master.RestoreServiceProviderLBDb()
+
+}
+
+// services: This function should be returning logical state instead of driver state
+func (d *daemon) services(id string) ([]core.State, error) {
+	var (
+		err error
+		svc *mastercfg.CfgServiceLBState
+	)
+
+	svc = &mastercfg.CfgServiceLBState{}
+	if svc.StateDriver, err = utils.GetStateDriver(); err != nil {
+		return nil, err
+	}
+
+	if id == "all" {
+		return svc.ReadAll()
+	} else if err := svc.Read(id); err == nil {
+		return []core.State{core.State(svc)}, nil
+	}
+
+	return nil, core.Errorf("Unexpected code path")
```

Comment:
Let's print and return the actual error here instead of "Unexpected code path" :)